chain/network/src/peer_manager/tests/sync_snapshot_hosts.rs

```diff
@@ -0,0 +1,282 @@
+        .await;
+
+    tracing::info!(target:"test", "Make sure that only the valid message is broadcast.");
+
+    let msg = peer2.events.recv_until(take_sync_snapshot_msg).await;
+    assert_eq!(msg.hosts, vec![info2]);
+}
+
+/// Test that SyncSnapshotHosts message is propagated between many PeerManager instances.
+/// Four peer managers are connected into a ring:
+/// [0] - [1]
+///  |     |
+/// [2] - [3]
+/// And then the managers propagate messages among themeselves.
+#[tokio::test]
+async fn propagate() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    // Adjust the file descriptors limit, so that we can create many connection in the test.
+    const MAX_CONNECTIONS: usize = 2;
+    let limit = rlimit::Resource::NOFILE.get().unwrap();
+    rlimit::Resource::NOFILE
+        .set(std::cmp::min(limit.1, (1000 + 4 * FDS_PER_PEER * MAX_CONNECTIONS) as u64), limit.1)
+        .unwrap();
+
+    tracing::info!(target:"test", "Create four peer manager instances.");
+    let mut pms = vec![];
+    for _ in 0..4 {
+        pms.push(
+            peer_manager::testonly::start(
+                clock.clock(),
+                near_store::db::TestDB::new(),
+                chain.make_config(rng),
+                chain.clone(),
+            )
+            .await,
+        );
+    }
+
+    tracing::info!(target:"test", "Connect the four peer managers into a ring.");
+    // [0] - [1]
+    //  |     |
+    // [2] - [3]
+    pms[0].connect_to(&pms[1].peer_info(), tcp::Tier::T2).await;
+    pms[0].connect_to(&pms[2].peer_info(), tcp::Tier::T2).await;
+    pms[1].connect_to(&pms[3].peer_info(), tcp::Tier::T2).await;
+    pms[2].connect_to(&pms[3].peer_info(), tcp::Tier::T2).await;
+
+    tracing::info!(target:"test", "Receive the inital empty snapshot info messages.");
+    // Each peer_manager connects to two others and receives two empty sync_snapshot messages
+    for _ in 0..2 {
+        for pm in &mut pms {
+            let empty_sync_msg = pm.events.recv_until(take_sync_snapshot_msg_manager).await;
+            assert_eq!(empty_sync_msg.hosts, vec![]);
+        }
+    }
+
+    tracing::info!(target:"test", "Send a SnapshotHostInfo message from peer manager #1.");
+    let info1 =
+        make_snapshot_host_info(&pms[1].peer_info().id, 123, vec![2, 3], &pms[1].cfg.node_key);
+
+    let message = PeerManagerMessageRequest::NetworkRequests(NetworkRequests::SnapshotHostInfo {
+        sync_hash: info1.sync_hash,
+        epoch_height: info1.epoch_height,
+        shards: info1.shards.clone(),
+    });
+
+    pms[1].actix.addr.send(message.with_span_context()).await.unwrap();
+
+    tracing::info!(target:"test", "Make sure that the message sent from #1 reaches #2 on the other side of the ring.");
+    let received_message = pms[2].events.recv_until(take_sync_snapshot_msg_manager).await;
```

Comment:
What if we moved this inside `peer_manager/testonly.rs` and actually checked the internal state of the SnapshotHostCache, along the lines of:
https://github.com/near/nearcore/blob/ed3dc7eb7b2237ea3215c816cfd6c687a7219a37/chain/network/src/peer_manager/testonly.rs#L404-L410

That is, verify that the event is not only received by the peer manager actor but actually processed correctly into the local state?